packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart

```diff
@@ -780,36 +850,178 @@ class AndroidWebViewWidget extends PlatformWebViewWidget {
           params,
           displayWithHybridComposition:
               _androidParams.displayWithHybridComposition,
+          platformViewsServiceProxy: _androidParams.platformViewsServiceProxy,
+          view:
+              (_androidParams.controller as AndroidWebViewController)._webView,
+          instanceManager: _androidParams.instanceManager,
+          layoutDirection: _androidParams.layoutDirection,
         )
           ..addOnPlatformViewCreatedListener(params.onPlatformViewCreated)
           ..create();
       },
     );
   }
+}
 
-  AndroidViewController _initAndroidView(
-    PlatformViewCreationParams params, {
-    required bool displayWithHybridComposition,
-  }) {
-    if (displayWithHybridComposition) {
-      return _androidParams.platformViewsServiceProxy.initExpensiveAndroidView(
-        id: params.id,
-        viewType: 'plugins.flutter.io/webview',
-        layoutDirection: _androidParams.layoutDirection,
-        creationParams: _androidParams.instanceManager.getIdentifier(
-            (_androidParams.controller as AndroidWebViewController)._webView),
-        creationParamsCodec: const StandardMessageCodec(),
-      );
-    } else {
-      return _androidParams.platformViewsServiceProxy.initSurfaceAndroidView(
-        id: params.id,
-        viewType: 'plugins.flutter.io/webview',
-        layoutDirection: _androidParams.layoutDirection,
-        creationParams: _androidParams.instanceManager.getIdentifier(
-            (_androidParams.controller as AndroidWebViewController)._webView),
-        creationParamsCodec: const StandardMessageCodec(),
-      );
-    }
+/// Object specifying creation parameters for creating a [AndroidCustomViewWidget].
+///
+/// When adding additional fields make sure they can be null or have a default
+/// value to avoid breaking changes.
+///
+/// The [AndroidCustomViewWidgetCreationParams] are used internally only to
+/// instantiate the [AndroidCustomViewWidget]. This class is visible for
+/// testing purposes only and should not be used externally.
+@immutable
+@visibleForTesting
+class AndroidCustomViewWidgetCreationParams {
```

Comment:
This class shouldn't be necessary since it is only accessed by package and tests. You should be able to put the fields into `AndroidCustomViewWidget`.

Reply:
Agreed, I have removed the `AndroidCustomViewWidgetCreationParams` class and updated the `AndroidCustomViewWidget.private` constructor to accept the required parameters. 

Since this is a private constructor and not called externally, I have removed the "optional" parameters and instead hardcoded the default values into the widget it self.